File: frontend/src/types/files.ts
export interface FileEntry {
  name: string
  path: string
  is_dir: boolean
  size: number | null
  modified: string | null
  children_count: number | null
}

export interface RenameRequest {
  path: string
  new_name: string
}

export interface MoveFileRequest {
  src: string
  dst: string
}

export interface CreateDirRequest {
  path: string
}

File: backend/app/schemas/files.py
"""File-manager schemas."""

from __future__ import annotations

from pydantic import BaseModel


class FileEntry(BaseModel):
    """One entry in a directory listing."""

    name: str
    path: str
    is_dir: bool
    size: int | None = None
    modified: str | None = None
    children_count: int | None = None


class RenameRequest(BaseModel):
    """Rename a file or directory."""

    path: str
    new_name: str


class MoveFileRequest(BaseModel):
    """Move or copy a file/directory."""

    src: str
    dst: str


class CreateDirRequest(BaseModel):
    """Create a new directory inside the sandbox."""

    path: str
